### PR TITLE
Add support for decoding different jpg subsampling factors

### PR DESCRIPTION
**Pull request by ram-mohan**
Initial versions supports decoding images with subsampling factor 4:2:0. This changes broadens the support.

Test: ./ultrahdr_unit_test
Test: ./ultrahdr_dec_fuzzer

### CODE REVIEW

File: lib/include/ultrahdr/jpegdecoderhelper.h [hunk 1]
```diff
@@ -38,139 +38,145 @@ extern "C" {
...
-  JpegDecoderHelper();
-  ~JpegDecoderHelper();
-  /*
-   * Decompresses JPEG image to raw image (YUV420planer, grey-scale or RGBA) format. After
-   * calling this method, call getDecompressedImage() to get the image.
-   * Returns false if decompressing the image fails.
-   */
-  bool decompressImage(const void* image, int length, decode_mode_t decodeTo = DECODE_TO_YCBCR);
-  /*
-   * Returns the decompressed raw image buffer pointer. This method must be called only after
```
Comment (DichenZhang1) [inline lib/include/ultrahdr/jpegdecoderhelper.h:71]: For all the similar notes, I think they are necessary. Could you add those to the new methods' comments?

**Review by DichenZhang1 — APPROVED** [review]

File: lib/include/ultrahdr/jpegdecoderhelper.h (hunk 1 above)
Comment (DichenZhang1) [reply lib/include/ultrahdr/jpegdecoderhelper.h:71]: I mean we need to keep "This method must be called only after ..." note

Comment (ram-mohan) [reply lib/include/ultrahdr/jpegdecoderhelper.h:71]: They are present dichen, I did not remove. I move them to one place.

Lines 102-103
/*! Below public methods are only effective if a call to decompressImage() is made and it returned
   * true. */
<CODE>
Lines 114-115
  /*! Below public methods are only effective if a call to parseImage() or decompressImage() is made
   * and it returned true. */

File: lib/include/ultrahdr/jpegdecoderhelper.h [hunk 2]
```diff
@@ -38,139 +38,145 @@ extern "C" {
...
-  PARSE_ONLY = 0,          // Dont decode. Parse for dimensions, EXIF, ICC, XMP
-  DECODE_TO_RGBA = 1,      // Parse and decode to rgba
-  DECODE_TO_YCBCR = 2,     // Parse and decode to YCBCR or Grayscale
-                           // if input has 1 channel, decode to Grayscale
-                           // if input has 3 channels, decode to YCBCR
-  DECODE_TO_GAIN_MAP = 3,  // parse and decode gain map.
-                           // if input has 1 channel, decode to Grayscale
-                           // if input has 3 channels, decode to RGBA
+  PARSE_STREAM = (1 << 0),   /**< Parse jpeg header, APPn markers (Exif, Icc, Xmp, Iso) */
+  DECODE_STREAM = (1 << 16), /**< Single channel images are decoded to YCbCr Color Space and multi
```
[the remark this answers is not in the document]
Comment (ram-mohan) [reply lib/include/ultrahdr/jpegdecoderhelper.h]: Done.

**Pull request merged**